unityfiles/Assets/Scripts/Core/SceneManager.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+
+/// <summary>
+/// The scene manager is responsible for loading scenes.
+/// The scene manager is a singleton, and there can only exist a single
+/// instane of it.
+/// It can load a scene by name and its build index.
+/// </summary>
+public sealed class SceneManager {
+    // Scenemanager instance
+    private static SceneManager instance = null;
+    // Lock for thread safety
+    private static readonly object padlock = new Object();
+
+    /// <summary>
+    /// Check if instance of SceneManager exist, if it does, return it.
+    /// Else lock the resource and create a new instance of it and return it.
+    /// </summary>
+    /// <value>SceneManager instance</value>
+    public static SceneManager Instance {
+        get {
+            {
+                lock(padlock) {
```

Comment:
Hva er dette? `lock()` Hvordan virker det og hvorfor bruke det?

Reply:
Er C# mekanisme for thread safety. Vist det av an grunn blir kjørt to threads og begge spør om instance av dette scriptet vil det forhindre at begge trådene genererer hver sin instance. Så den sørger for at kun en thread har tilgang avgangen, og vist en thread genererer instansen, vil den andre da ha tilgang til den genererte instansen.